### PR TITLE
Resource: add udev based Sysfs GPIO support

### PR DESCRIPTION
**Pull request by jeromebrunet**
This PR adds support for GPIOs for which the gpiochip is identified using udev.

This is useful when the probe order of the gpiochip is not predictable. 
In this case, this is case it is not possible to know the global gpio number required by SysfsGPIO resource.

With MatchedSysfsGPIO, the pin number is local to the gpiochip and gpiochip is identified with udev

**Checklist**
- [x] Documentation for the feature
- [x] Tests for the feature : No obvious test for udev based resources ?

- [x] The arguments and description in doc/configuration.rst have been updated

- [x] PR has been tested

### CODE REVIEW

**Review by jluebbe — CHANGES_REQUESTED** [review]
Looks good otherwise.

File: doc/configuration.rst [hunk 1]
```diff
@@ -523,6 +523,48 @@ NetworkHIDRelay
...
+The gpiochip is identified by matching udev properties. This allows
+identification through hot-plugging or rebooting for controllers like
+USB based gpiochips.
+
+.. code-block:: yaml
+
+   MatchedSysfsGpio:
+     match:
+       - '@SUBSYSTEM': 'usb'
+       - '@ID_SERIAL_SHORT': 'D38EJ8LF'
```
[the remark this answers is not in the document]
Comment (jluebbe) [reply doc/configuration.rst]: I've force-pushed this change.

**Comment by jluebbe** [comment]
@jeromebrunet Are you fine with my fixup?

File: doc/configuration.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeromebrunet) [reply doc/configuration.rst]: Oh yes - I forgot to change this in the doc - sorry

**Comment by jeromebrunet** [comment]
> @jeromebrunet Are you fine with my fixup?

Yes - Thanks for doing it

**Review by jluebbe — APPROVED** [review]

**Pull request merged**